File: 剑指offer刷题/剑指Offer-Python实现/5_PushPop.py
# 用两个栈来实现一个队列，完成队列的Push和Pop操作。 队列中的元素为int类型。

# -*- coding:utf-8 -*-
stack1 = []
stack2 = []
class Solution:
    def push(self, node):
        # write code here
        stack1.append(node)
    def pop(self):
        # return xx
        if(len(stack2)==0):
            while(len(stack1)>0):
                stack2.append(stack1.pop())
        return stack2.pop()

File: SwordToOffer/SwordToOffer-JavaScriptSolution/5_PushPop.js
// 用两个栈来实现一个队列，完成队列的Push和Pop操作。 队列中的元素为int类型。

// 栈先进后出，队列先进先出
const stack1 = []
const stack2 = []

function push(node) {
    // write code here
    stack1.push(node)
}

function pop() {
    // write code here
    if (stack2.length === 0) {
        while (stack1.length > 0) {
            stack2.push(stack1.pop())
        }
    }
    return stack2.pop()
}